torch/onnx/_internal/fx/op_validation.py

```diff
@@ -0,0 +1,224 @@
+            )
+            diagnostic = diagnostics.export_context().inflight_diagnostic()
+            diagnostic.with_additional_message(
+                f"### Op level debug fails on PyTorch\n"
+                f"{diagnostics.decorator.format_exception_in_markdown(runtime_error)}"
+            )
+            diagnostic.level = diagnostics.levels.ERROR
+            return
+
+        # TODO(titaiwang): Need Opschema from ONNX function to better split args/kwargs
+        # Currently, we only support torch.Tensor to numpy array. Potentially, we
+        # could fail on INT64. However, we don't support dims/indices INT64 validation.
+        input_onnx = [
+            onnx_proto_utils._convert_tensor_to_numpy(x)
+            if isinstance(x, (torch.Tensor, torch.dtype, list, tuple))
+            else x
+            for x in torch_args
+        ]
+        kwargs_onnx = fx_to_onnxscript.filter_incompatible_and_dtype_convert_kwargs(
+            torch_kwargs
+        )
+        try:
+            ort_outputs = symbolic_fn(*input_onnx, **kwargs_onnx)
+        except ValueError as value_error:
+            # FIXME(titaiwang): This is caused by wronly split args/kwargs.
+            # When Opschema is ready, we should follow Opschema to split args/kwargs.
+            warnings.warn(
+                f"\nBypass the test of running on ONNX Op {function_name} with "
+                f"ValueError: \n{value_error}.\n This is possibly raised by "
+                f"unsupported input args due to lack of Opschema.\n"
+            )
+            diagnostic = diagnostics.export_context().inflight_diagnostic()
+            diagnostic.with_additional_message(
+                f"### Op level debug is bypassed\n"
+                f"{diagnostics.decorator.format_exception_in_markdown(value_error)}"
+            )
+            diagnostic.level = diagnostics.levels.WARNING
+            return
+        except RuntimeError as runtime_error:
+            warnings.warn(
+                f"\nFail the test of running on ONNX Op {function_name} with "
+                f"RuntimeError: \n{runtime_error}.\n"
+            )
+            diagnostic = diagnostics.export_context().inflight_diagnostic()
+            diagnostic.with_additional_message(
+                f"### Op level debug fails on ONNXRUNTIME:\n"
+                f"{diagnostics.decorator.format_exception_in_markdown(runtime_error)}"
+            )
+            diagnostic.level = diagnostics.levels.ERROR
+            return
+
+        flattened_torch_outputs, _ = _pytree.tree_flatten(expected_outputs)
+        flattened_function_outputs, _ = _pytree.tree_flatten(ort_outputs)
+
+        assert flattened_torch_outputs
+        assert len(flattened_torch_outputs) == len(flattened_function_outputs)
+
+        for torch_output, function_output in zip(
+            flattened_torch_outputs, flattened_function_outputs
+        ):
+            try:
+                if not isinstance(function_output, np.ndarray):
```

Comment:
nit: instead check that `function_output` is "onnxscript tensor" type?

Reply:
In fact, this should be deleted.

Reply:
Done